python_modules/dagster/dagster/_core/pipes/subprocess.py

```diff
@@ -93,17 +96,17 @@ def run(
         """
         with open_pipes_session(
             context=context,
-            context_injector=self.context_injector,
-            message_reader=self.message_reader,
+            context_injector=self._context_injector,
+            message_reader=self._message_reader,
             extras=extras,
         ) as pipes_session:
             process = Popen(
                 command,
-                cwd=cwd or self.cwd,
+                cwd=cwd or self._cwd,
                 env={
-                    **pipes_session.get_bootstrap_env_vars(),
-                    **self.env,
+                    **(self._base_env if self._base_env is not None else os.environ),
```

Comment:
this is currently "inherit or use the specific values set on the client" 

the main downside here is if you want to set some env vars at the client and inherit you have to do `env={ **os.environ, 'set': 'other_vals'}`

looks like for the old shell op we ended up always inheriting https://github.com/dagster-io/dagster/pull/6513

so some alternative options are:
* have a separate (default on) arg control whether we inherit process env vars
* just always inherit and leave it to direct `open_pipes_session` usage to blank out

Reply:
Re: 

> the main downside here is if you want to set some env vars at the client and inherit you have to do env={ **os.environ, 'set': 'other_vals'}

If this were ok why not tell people just to do `env=os.environ`?

Unless I am confuse someone. Seems like to me we should also inherit os.environ or not and then us the "just always inherit and leave it to direct open_pipes_session usage to blank out": approach.